packages/zoltan2/core/src/environment/Zoltan2_MachineDragonflyRCAForTesting.hpp

```diff
@@ -481,11 +474,13 @@ class MachineDragonflyRCAForTesting : public Machine <pcoord_t, part_t> {
     return false;
 #endif
 */
-    srand(this->myRank);
+//    srand(this->myRank);
+
+    int large_prime = 1117;  
 
-    int x = rand() % 11;
-    int y = rand() % 6;
-    int z = rand() % 16;
+    int x = this->myRank * large_prime % 11;
+    int y = this->myRank * large_prime % 6;
+    int z = this->myRank * large_prime % 16;
```

Comment:
Does this change give every processor the same (x,y,z) coordinate in the test?  Is that what you want?